openmls/src/group/core_group/process.rs

```diff
@@ -93,25 +95,29 @@ impl CoreGroup {
         proposal_store: &ProposalStore,
         own_kpbs: &[KeyPackageBundle],
         backend: &impl OpenMlsCryptoProvider,
-    ) -> Result<ProcessedMessage, CoreGroupError> {
+    ) -> Result<ProcessedMessage, UnverifiedMessageError> {
         // Add the context to the message and verify the membership tag if necessary.
         // If the message is older than the current epoch, we need to fetch the correct secret tree first.
-        let message_secrets = self.message_secrets_mut(unverified_message.epoch())?;
+        let message_secrets = self
+            .message_secrets_mut(unverified_message.epoch())
+            .map_err(|_| UnverifiedMessageError::NoPastEpochData)?;
```

Comment:
Don't we have these per-function error types such that we can actually map on the thrown error, rather than do "blind" `map_err` to a fixed error? The benefit of actually matching on the error would be that, should we refactor the function in the future, the type checker will make sure that callers properly handle any new errors. (Same for the other `map_err`s below.)

Or is this just a stop-gap, because the underlying function doesn't have a per-function error yet?